plugins/csp-web-api/src/Plugin.cpp

```diff
@@ -466,18 +474,38 @@ void Plugin::update() {
         }
 
       } else {
-        // Capturing color images is pretty straight-forward.
-        std::vector<std::byte> capture(mCaptureWidth * mCaptureHeight * 3);
-        glReadPixels(0, 0, mCaptureWidth, mCaptureHeight, GL_RGB, GL_UNSIGNED_BYTE, &capture[0]);
-
-        if (mCaptureFormat == "tiff") {
-          tiffWriteToVector(mCapture, capture, mCaptureWidth, mCaptureHeight, 3, 8);
-        } else if (mCaptureFormat == "png") {
-          stbi_write_png_to_func(&stbWriteToVector, &mCapture, mCaptureWidth, mCaptureHeight, 3,
-              capture.data(), mCaptureWidth * 3);
+        if (mCaptureFormat == "raw") {
+          // larger output size, but performance is comparable or better because there is no image
+          // encoding
+          mCapture.resize(mCaptureWidth * mCaptureHeight * 3 * sizeof(float));
+          if (mAllSettings->mGraphics.pEnableHDR.get()) {
+            // using the hdr buffer
+            std::shared_ptr<cs::graphics::HDRBuffer> hdrBuffer = mGraphicsEngine->getHDRBuffer();
+            VistaTexture* luminance_buffer = hdrBuffer->getCurrentWriteAttachment();
+            luminance_buffer->Bind();
+            glGetTexImage(
+                luminance_buffer->GetTarget(), 0, GL_RGB, GL_FLOAT, (void*)mCapture.data());
```

Comment:
We usually try to avoid C-style casts.
```suggestion
            glGetTexImage(luminance_buffer->GetTarget(), 0, GL_RGB, GL_FLOAT,
                static_cast<void*>(mCapture.data()));
```